proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/VSPProxyTranslateResponseTest.java

```diff
@@ -0,0 +1,149 @@
+    public static final DropwizardClientRule testVspServerErrorClientRule = new DropwizardClientRule(new TestVSPServerErrorResource());
+
+    @ClassRule
+    public static final DropwizardClientRule testVspClientErrorClientRule = new DropwizardClientRule(new TestVSPClientErrorResource());
+
+    @Spy
+    JsonClient jsonClient = new JsonClient(
+            new ErrorHandlingClient(ClientBuilder.newClient()),
+            new JsonResponseProcessor(new ObjectMapper())
+    );
+
+    @Test
+    public void shouldReturnTranslatedHubResponseAuthnFailed() {
+
+        VerifyServiceProviderProxy vspProxy = new VerifyServiceProviderProxy(jsonClient, testVspClientRule.baseUri());
+
+        TranslatedHubResponse response =
+                vspProxy.getTranslatedHubResponse(
+                        new VerifyServiceProviderTranslationRequest(null, "_1234", "LEVEL_2")
+                );
+
+        assertEquals(VspScenario.AUTHENTICATION_FAILED, response.getScenario());
+        assertEquals("pid1234", response.getPid());
+        assertEquals(VspLevelOfAssurance.LEVEL_2, response.getLevelOfAssurance());
+        assertEquals(null, response.getAttributes());
+
+        Mockito.verify(jsonClient).post(
+                Mockito.argThat((VerifyServiceProviderTranslationRequest request) -> request.getLevelOfAssurance().equals("LEVEL_2")),
+                eq(UriBuilder.fromUri(String.format("%s/translate-response", testVspClientRule.baseUri())).build()),
+                eq(TranslatedHubResponse.class)
+        );
+    }
+
+    @Test
+    public void shouldReturnTranslatedHubResponse() {
+
+        VerifyServiceProviderProxy vspProxy = new VerifyServiceProviderProxy(jsonClient, testVspClientRule.baseUri());
+
+        TranslatedHubResponse response =
+                vspProxy.getTranslatedHubResponse(
+                        new VerifyServiceProviderTranslationRequest("SAMLResponse1234", "_1234", "LEVEL_2")
+                );
+
+        assertEquals(VspScenario.IDENTITY_VERIFIED, response.getScenario());
+        assertEquals("123456", response.getPid());
+        assertEquals(VspLevelOfAssurance.LEVEL_2, response.getLevelOfAssurance());
+    }
+
+    @Test(expected = ApplicationException.class)
+    public void shouldThrowApplicationExceptionOnClientError() {
+        VerifyServiceProviderProxy vspProxy = new VerifyServiceProviderProxy(jsonClient, testVspClientErrorClientRule.baseUri());
+
+        TranslatedHubResponse response =
+                vspProxy.getTranslatedHubResponse(
+                        new VerifyServiceProviderTranslationRequest("SAMLResponse1234", "_1234", "LEVEL_2")
+                );
+        fail("Expected exception not thrown");
+    }
+
+    @Test(expected = ApplicationException.class)
+    public void shouldThrowApplicationExceptionOnServerError() {
+        VerifyServiceProviderProxy vspProxy = new VerifyServiceProviderProxy(jsonClient, testVspServerErrorClientRule.baseUri());
+
+        TranslatedHubResponse response =
```

Comment:
`response` is not used - can remove the variable declaration

Reply:
Removed.